map/autoware_lanelet2_map_validator/test/src/test_intersection_area_dangling_reference.cpp

```diff
@@ -0,0 +1,83 @@
+// Copyright 2024 Autoware Foundation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "lanelet2_map_validator/validators/intersection/intersection_area_dangling_reference.hpp"
+#include "map_validation_tester.hpp"
+
+#include <gtest/gtest.h>
+#include <lanelet2_core/LaneletMap.h>
+
+#include <string>
+
+class TestIntersectionAreaDanglingReferenceError : public MapValidationTester
+{
+protected:
+  void SetUp() override
+  {
+    // prepare `map_` and `loading_errors_`
+    // this class uses erroneous map
+    load_target_map("intersection/intersection_area_with_dangling_reference.osm");
+  }
+
+private:
+};
+
+class TestIntersectionAreaDanglingReferenceOK : public MapValidationTester
+{
+protected:
+  void SetUp() override
+  {
+    // prepare `map_` and `loading_errors_`
+    // this class uses valid map
+    load_target_map("intersection/basic_intersection_area.osm");
+  }
+
+private:
+};
```

Comment:
I think only one class is enough and try loading the maps at the start of the test functions if there's no special reason.

You can write the validation subject on the second argument of `TEST_F`.